sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -333,6 +333,42 @@ impl TypedIntrinsicFunctionKind {
                     type_id,
                 )
             }
+            Intrinsic::AddrOf => {
+                if arguments.len() != 1 {
+                    errors.push(CompileError::IntrinsicIncorrectNumArgs {
+                        name: kind.to_string(),
+                        expected: 1,
+                        span,
+                    });
+                    return err(warnings, errors);
+                }
+                let ctx = ctx
+                    .with_help_text("")
+                    .with_type_annotation(insert_type(TypeInfo::Unknown));
+                let exp = check!(
+                    TypedExpression::type_check(ctx, arguments[0].clone()),
+                    return err(warnings, errors),
+                    warnings,
+                    errors
+                );
+                let copy_ty = resolve_type(exp.return_type, &span).unwrap().is_copy_type();
+                if copy_ty {
+                    errors.push(CompileError::IntrinsicUnsupportedArgType {
```

Comment:
I can see down in the test below that this just says 'Unsupported argument type to intrinsic "addr_of"' which would be better if it actually said 'Intrinsic "addr_of" cannot be passed a non-reference type."' or similar.

OTOH, we only expect this to be called by the library and library devs so verbosity isn't the highest priority.  Is there an easy way to add a hint to `IntrinsicUnsupportedArgType`, like an `Option<str>` arg or something?

Reply:
Formatting the `Option` here could be tricky. Let me try doing it.